internal/ecgit/ecgit.go

```diff
@@ -35,6 +35,39 @@ import (
 	"golang.org/x/oauth2"
 
 	"github.com/hacbs-contract/ec-cli/internal/utils"
+	e "github.com/hacbs-contract/ec-cli/pkg/error"
+)
+
+var (
+	EC001 = e.NewError("EC001", "", e.ErrorExitStatus) // TODO: add message and set the exit status
+	EC002 = e.NewError("EC002", "", e.ErrorExitStatus) // TODO: add message and set the exit status
+	EC003 = e.NewError("EC003", "", e.ErrorExitStatus) // TODO: add message and set the exit status
+	EC004 = e.NewError("EC004", "", e.ErrorExitStatus) // TODO: add message and set the exit status
+	EC005 = e.NewError("EC005", "", e.ErrorExitStatus) // TODO: add message and set the exit status
+	EC006 = e.NewError("EC006", "", e.ErrorExitStatus) // TODO: add message and set the exit status
+	EC007 = e.NewError("EC007", "", e.ErrorExitStatus) // TODO: add message and set the exit status
+	EC008 = e.NewError("EC008", "", e.ErrorExitStatus) // TODO: add message and set the exit status
+	EC009 = e.NewError("EC009", "", e.ErrorExitStatus) // TODO: add message and set the exit status
+	EC010 = e.NewError("EC010", "", e.ErrorExitStatus) // TODO: add message and set the exit status
+	EC011 = e.NewError("EC011", "", e.ErrorExitStatus) // TODO: add message and set the exit status
+	EC012 = e.NewError("EC012", "", e.ErrorExitStatus) // TODO: add message and set the exit status
+	EC013 = e.NewError("EC013", "", e.ErrorExitStatus) // TODO: add message and set the exit status
+	EC014 = e.NewError("EC014", "", e.ErrorExitStatus) // TODO: add message and set the exit status
+	EC015 = e.NewError("EC015", "", e.ErrorExitStatus) // TODO: add message and set the exit status
+	EC016 = e.NewError("EC016", "", e.ErrorExitStatus) // TODO: add message and set the exit status
+	EC017 = e.NewError("EC017", "", e.ErrorExitStatus) // TODO: add message and set the exit status
+	EC018 = e.NewError("EC018", "", e.ErrorExitStatus) // TODO: add message and set the exit status
+	EC019 = e.NewError("EC019", "", e.ErrorExitStatus) // TODO: add message and set the exit status
+	EC020 = e.NewError("EC020", "", e.ErrorExitStatus) // TODO: add message and set the exit status
+	EC021 = e.NewError("EC021", "", e.ErrorExitStatus) // TODO: add message and set the exit status
+	EC022 = e.NewError("EC022", "", e.ErrorExitStatus) // TODO: add message and set the exit status
+	EC023 = e.NewError("EC023", "", e.ErrorExitStatus) // TODO: add message and set the exit status
+	EC024 = e.NewError("EC024", "", e.ErrorExitStatus) // TODO: add message and set the exit status
+	EC025 = e.NewError("EC025", "", e.ErrorExitStatus) // TODO: add message and set the exit status
+	EC026 = e.NewError("EC026", "", e.ErrorExitStatus) // TODO: add message and set the exit status
+	EC027 = e.NewError("EC027", "", e.ErrorExitStatus) // TODO: add message and set the exit status
+	EC028 = e.NewError("EC028", "", e.ErrorExitStatus) // TODO: add message and set the exit status
+	EC029 = e.NewError("EC029", "", e.ErrorExitStatus) // TODO: add message and set the exit status
```

Comment:
So ugly though. Could we maybe generate a code from file and line number?

Reply:
Is it going to be tedious to main a message that is far away from the place where it's used?

Reply:
> So ugly though. Could we maybe generate a code from file and line number?

The linter has no notion of aesthetics, but a person could take this and refactor as they see it. I do see errors declared at the top of the file in other folk's golang code, so I guess it could be a convention folk use. Folk also tend to optimize the allocation by pre-allocating the errors this way (not sure if we'd care at all).

I don't think file/line numbers for error codes are not a good idea as they would be difficult with different versions and code changes to link up. Those could be in the message or `pkg/error.Error` could be extended to include those separately.

Linter finds all occurrences where we are returning a raw error and generates a separate `pkg/error.Error` code for each of those, we don't need to have that many errors, might not be an issue with 3-5 errors per file. Though I would argue for having many errors, e.g. say we have the same error code for "access denied" error and reused it in three places in a package for accessing three different resources, it might be difficult to reason about accessing what resource causes the error.

There is a pattern in ODBC/SQL errors where each error code has a reason code associated with it as well, though I think that would make it a bit more difficult to maintain.

Reply:
Ack, thanks. I have no objections, just thinking it through.